File: video-editor/models/messages/video_editing_step_changed.py
from typing import Literal

from models.broker_model import BrokerModel

EditingStepStatus = Literal["in_progress", "completed", "failed"]


class VideoEditingStepChanged(BrokerModel):
    video_id: str
    node_id: str
    user_id: str
    step: str
    status: EditingStepStatus
    error: str | None = None

File: websocket-gateway/src/Contracts/VideoEditor/VideoEditingStepChanged.cs
namespace WebSocketGateway.Contracts.VideoEditor;

public class VideoEditingStepChanged
{
    public required string VideoId { get; set; }

    public required string NodeId { get; set; }

    public required string UserId { get; set; }

    public required string Step { get; set; }

    public required string Status { get; set; }

    public string? Error { get; set; }
}
